server/src/main/java/org/opensearch/wlm/Rule.java

```diff
@@ -0,0 +1,370 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.wlm;
+
+import org.opensearch.cluster.AbstractDiffable;
+import org.opensearch.cluster.Diff;
+import org.opensearch.common.annotation.ExperimentalApi;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParseException;
+import org.opensearch.core.xcontent.XContentParser;
+import org.joda.time.Instant;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+import static org.opensearch.cluster.metadata.QueryGroup.isValid;
+
+/**
+ * Class to define the Rule schema
+ * {
+ *     "_id": "fwehf8302582mglfio349==",
+ *     "index_pattern": ["logs123", "user*"],
+ *     "query_group": "dev_query_group_id",
+ *     "updated_at": "01-10-2025T21:23:21.456Z"
+ * }
+ */
+@ExperimentalApi
+public class Rule extends AbstractDiffable<Rule> implements ToXContentObject {
+    private final String _id;
+    private final Map<RuleAttribute, Set<String>> attributeMap;
+    private final Feature feature;
+    private final String label;
+    private final String updatedAt;
+    public static final Map<Feature, Set<RuleAttribute>> featureAlloedAttributesMap = Map.of(
```

Comment:
[NIT] typo in the constant name. we should name the constants in uppercase to distinguish it from variables.